### PR TITLE
Refactor parse_title_format

### PR DESCRIPTION
**Pull request by RyanDwyer**
Don't read properties until needed, and do it with less lines of code.

### CODE REVIEW

File: sway/tree/view.c [hunk 1]
```diff
@@ -641,6 +641,14 @@ void view_child_destroy(struct sway_view_child *child) {
 	}
 }
 
+static size_t append_prop(char *buffer, const char *value) {
```
Comment (RedSoxFan) [inline sway/tree/view.c]: Would it be worth moving this to `common/stringop.c` and renaming it to something like `lenient_strcat_size`? I would guess that this is not the only place that could make use of it.

Comment (RyanDwyer) [reply sway/tree/view.c]: My preference is to move it only if there's anything else that can make use of it. So I'd leave it where it is until that happens.

Comment (RedSoxFan) [reply sway/tree/view.c]: Ok. That works

**Comment by emersion** [comment]
Thanks!

**Pull request merged**